control-plane/csi-driver/src/context.rs

```diff
@@ -173,10 +205,114 @@ impl TryFrom<&HashMap<String, String>> for CreateParams {
             None => 1,
         };
 
+        let volume_group = Parameters::volumne_group(args.get(Parameters::VolumeGroup.as_ref()))
+            .map_err(|_| {
+                tonic::Status::invalid_argument(
+                    "Invalid `openebs.io/volumegroup` value, expected a bool",
+                )
+            })?;
+
+        let volume_group_name = if volume_group.unwrap_or(false) {
+            generate_volume_group_name(
+                args.get(Parameters::PvcName.as_ref()).cloned(),
+                args.get(Parameters::PvcNamespace.as_ref()).cloned(),
+            )
+        } else {
+            None
+        };
+
         Ok(Self {
             publish_params,
             share_protocol,
             replica_count,
+            volume_group: volume_group_name,
         })
     }
 }
+
+// Generate a volume group name from the parameters.
+// 1. Both pvc name and ns should be valid.
+// 2. Pvc name should follow the sts pvc naming convention.
+fn generate_volume_group_name(pvc_name: Option<String>, pvc_ns: Option<String>) -> Option<String> {
+    match (pvc_name.clone(), pvc_ns.clone()) {
```

Comment:
why clone here, can't we just take reference? we're using it below as a reference..

Reply:
Changed the parameters to take `&Option<String>`